### DISCUSSION TITLE
Donations

### DISCUSSION THREAD

**Discussion by skbolton** [zk-org/zk#26] [General]
I have been thinking about building my own wiki/zettelkasten style note taking system. This project checks so many of the boxes of things I would want. It also is very organized for such a young project. How can I donate?

**Comment by mickael-menu** [comment]
Welcome Stephen!

I really appreciate your offer, but I think it would be too much of a (tax) hassle for me to set up donations at this point (I'm based in France). The best way to help `zk` is to contribute:

* Directly on the codebase. [Go is a really simple language](https://tour.golang.org/), I actually learned it for this project. Open an issue first to talk about the stuff you'd like to implement.
* Through detailed and reproducible bug reports.
* On improving support in text editors:
    * There is some on-going work to [make `zk` compatible with LSP clients](https://github.com/mickael-menu/zk/issues/22). However it can only go so far feature-wise, but some LSP clients can be extended with JavaScript plugins (VS Code, coc.nvim).
    * @megalithic started [`zk.nvim`](https://github.com/megalithic/zk.nvim), a more advanced Lua plugin to integrate `zk` in Neovim.
* By opening more general discussions on the best formats/syntaxes and ways to manage a plain text Zettelkasten.

What are the features you think `zk` is missing at the moment?

**Comment by skbolton** [comment]
Sounds good. I am definitely interested in the nvim plugin and will be able to contribute there. I could see that plugin replacing what I use vim-zettel for

**Reply by megalithic** [reply]
Would love to have the contributions! I’ve got some architectural changes I am about to make to clean up a lot of things. I’ll start documenting some of those thoughts and start a projects page for that repo. @mickael-menu has done 99% of the work with zk and the lsp functionality already!
